src/b3d/modeling_utils.py

```diff
@@ -74,6 +76,39 @@ def logpdf(v, *args, **kwargs):
 bernoulli = tfp_distribution(lambda logits: tfp.distributions.Bernoulli(logits=logits))
 normal = tfp_distribution(tfp.distributions.Normal)
 
+
+###
+
+
+@Pytree.dataclass
+class RenormalizedLaplace(genjax.ExactDensity):
+    def sample(self, key, loc, scale, low, high):
+        warnings.warn(
+            "RenormalizedLaplace sampling is currently not implemented correctly."
+        )
+        x = tfp.distributions.Laplace(loc, scale).sample(seed=key)
+        return jnp.clip(x, low, high)
+
+    def logpdf(self, obs, loc, scale, low, high):
+        laplace_logpdf = tfp.distributions.Laplace(loc, scale).log_prob(obs)
+        p_below_low = tfp.distributions.Laplace(loc, scale).cdf(low)
+        p_below_high = tfp.distributions.Laplace(loc, scale).cdf(high)
+        log_integral_of_laplace_pdf_over_this_range = jnp.log(
+            p_below_high - p_below_low
+        )
```

Comment:
For numerical stability, perhaps we can consider using [`Laplace.log_cdf`](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Laplace#log_cdf) and [`Laplace. log_survival_function`](https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/Laplace#log_survival_function)?

(btw also just for clarity: I think `log_integral_of_laplace_pdf_over_this_range` is actually referring to the integral of laplace _within_ the range?)

Reply:
@horizon-blue can you please suggest the right code snippet we need to do `log_cdf` and `log_survival_function` here?